libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -1854,6 +1854,34 @@ MAV_RESULT GCS_MAVLINK::handle_command_camera(const mavlink_command_long_t &pack
     return result;
 }
 
+// sets ekf_origin if it has not been set.
+//  should only be used when there is no GPS to provide an absolute position
+void GCS_MAVLINK::set_ekf_origin(const Location& loc)
+{
+    // check location is valid
+    if (!check_latlng(loc)) {
+        return;
+    }
+
+    AP_AHRS &ahrs = AP::ahrs();
+
+    // check if EKF origin has already been set
+    Location ekf_origin;
+    if (ahrs.get_origin(ekf_origin)) {
```

Comment:
Should we really do this? I feel like this should be up to AHRS/EKF to decide.

Reply:
I don't know.  If we do change it then 6 months later we'll probably add a force option.

Reply:
You mean... if we **keep it** then 6 months later we'll probably add a force option.

Don't forget this is quite new code so this isn't exactly assumed behavior.